Src/Newtonsoft.Json/Linq/JToken.cs

```diff
@@ -2113,60 +2113,47 @@ public static JToken ReadFrom(JsonReader reader, JsonLoadSettings settings)
 
             if (reader.TokenType == JsonToken.None)
             {
-                if (!reader.Read())
+                if (!(settings != null && settings.CommentHandling == CommentHandling.Ignore ? reader.ReadAndSkipComments() : reader.Read()))
```

Comment:
Hmmm, I'm not a huge fan of calling methods that have a side effect inside a ternary statement, but I'm also not a fan of duplicating the throw.